projects/packages/my-jetpack/_inc/components/product-interstitial/index.jsx

```diff
@@ -154,13 +156,18 @@ export default function ProductInterstitial( {
 
 						// If no purchase is needed, redirect the user to the product screen.
 						if ( ! needsPurchase ) {
-							if ( postActivationUrl ) {
-								window.location.href = postActivationUrl;
-								return;
-							}
+							// for free products, we still initiate the site connection
+							handleRegisterSiteOnly().then( () => {
+								if ( postActivationUrl ) {
+									window.location.href = postActivationUrl;
```

Comment:
Also, if you pass `redirectUri` as a property to `useConnection`, you'll be able to achieve a similar result to this one.

Reply:
The only thing I am unsure about here is not knowing what the product or post-activation URL will be until this callback function is called 🤔 

I don't think I'd be able to know which URL to use when calling the `useConnection` hook?

Reply:
Doy. Just kidding. I can totally get this info earlier 😅